src/scripts/main.js

```diff
@@ -1,5 +1,48 @@
 'use strict';
 
-document.addEventListener('click', (e) => {
-  // write code here
+document.addEventListener('DOMContentLoaded', () => {
+  const wall = document.querySelector('.wall');
+  const spider = document.querySelector('.spider');
+
+  // eslint-disable-next-line no-shadow
+  wall.addEventListener('click', (event) => {
+    const wallRect = wall.getBoundingClientRect();
+    const spiderRect = spider.getBoundingClientRect();
+
+    const spiderHalfWidth = spiderRect.width / 2;
+    const spiderHalfHeight = spiderRect.height / 2;
+    const borderWidth = 10;
+
+    // const clickX = event.clientX;
+    // const clickY = event.clientY;
+
+    const clickX = event.clientX - wallRect.left;
+    const clickY = event.clientY - wallRect.top;
+
+    let newLeft = clickX - spiderHalfWidth - borderWidth;
+    let newTop = clickY - spiderHalfHeight - borderWidth;
+
+    // let newLeft = clickX - wallRect.left - spiderHalfWidth;
+    // let newTop = clickY - wallRect.top - spiderHalfHeight;
+
+    if (newLeft < 0) {
+      newLeft = 0;
+    }
+
+    if (newTop < 0) {
+      newTop = 0;
+    }
+
+    if (newLeft + spiderRect.width > wallRect.width - 2 * borderWidth) {
+      newLeft = wallRect.width - spiderRect.width - 2 * borderWidth;
+    }
+
+    if (newTop + spiderRect.height > wallRect.height - 2 * borderWidth) {
+      newTop = wallRect.height - spiderRect.height - 2 * borderWidth;
```

Comment:
The conditions for adjusting `newLeft` and `newTop` to ensure the spider stays within the wall's boundaries include `2 * borderWidth`. If the border width is not part of the wall's dimensions, this might cause incorrect boundary checks. Verify if the border width should be included in these calculations.